distribution/lib/Standard/Table/0.0.0-dev/src/Delimited/Delimited_Format.enso

```diff
@@ -0,0 +1,121 @@
+
+       If a row does not match the first row's column count, the function raises
+       an `Invalid_Row`. If a quote is opened and never closed, a
+       `Mismatched_Quote` warning occurs.
+
+       Arguments:
+       - delimiter: The delimiter character to split the file into columns. An
+         `Illegal_Argument_Error` error is returned if this is an empty string.
+       - encoding: The encoding to use when reading the file.
+       - skip_rows: The number of rows to skip from the top of the file.
+       - row_limit: The maximum number of rows to read from the file. This count
+         does not include the header row (if applicable).
+       - quote_style: Specifies the style of quotes used for reading and
+         writing.
+       - headers: If set to `True`, the first row is used as column names. If
+         set to `False`, the column names are generated by adding increasing
+         numeric suffixes to the base name `Column` (i.e. `Column_1`,
+         `Column_2` etc.). If set to `Infer`, the process tries to infer if
+         headers are present on the first row. If the column names are not
+         unique, numeric suffixes will be appended to disambiguate them.
+       - value_formatter: Formatter to parse text values into numbers, dates,
+         times, etc. If `Nothing` values are left as Text.
+       - keep_invalid_rows: Specifies whether rows that contain less or more
+         columns than expected should be kept (setting the missing columns to
+         `Nothing` or dropping the excess columns) or dropped.
+       - line_endings: Sets the line ending style to use. Defaults to `Infer` -
+         when reading a file or appending to an existing file, the line endings
+         are detected from file contents; when writing a new file in `Infer`
+         mode the `Unix` line endings are used.
+       - comment_character: Sets the character which indicates the start of a
+         comment within a delimited file. Any line that begins with the comment
+         character is skipped. The comment character is treated as any other
+         character if it anywhere else than at the beginning of the line. This
+         option is only applicable for read mode and does not affect writing. It
+         defaults to `Nothing` which means that comments are disabled.
+    Delimited (delimiter:Text) (encoding:Encoding=Encoding.utf_8) (skip_rows:Integer=0) (row_limit:Integer|Nothing=Nothing) (quote_style:Quote_Style=Quote_Style.With_Quotes) (headers:Boolean|Infer=Infer) (value_formatter:Data_Formatter|Nothing=Data_Formatter_Data) (keep_invalid_rows:Boolean=True) (line_endings:Line_Ending_Style=Infer) (comment_character:Text|Nothing=Nothing)
+
+    ## If the File_Format supports reading from the file, return a configured instance.
+    can_read : File -> Any
+    can_read self file =
+        case file.extension of
+            ".csv" -> Delimited ','
+            ".tsv" -> Delimited '\t'
+            _ -> Nothing
```

Comment:
I'm not sure about this.

Shouldn't we instead do it so that `Delimited ','` supports `csv` and `Delimited '\t'` supports `tsv` - and just put two instances of these in our supported types list?
It seems a bit weird to me that we construct `Delimited ','` but then it answers positively to `can_read "foo.tsv"` but will yield a changed configuration. Shouldn't we instead just have these as two kind-of-separate file types? And maybe then we could get rid of this pre-configuration logic and just make `can_read` return a `Boolean`?

Unless you envision this more complex logic should prove more useful in the future - which I think is quite possible indeed. Still I'm a bit uneasy about the fact that we need to instantiate an essentially fully configured file type (like `Delimited`) instance to then infer a possibly another instance. Feels to me like mixing responsibilities a bit - it seems that we may need to have two separate concepts - one for detecting the default configuration for a given set of formats and another for holding a concrete configuration after it was detected or configured manually. So maybe this `can_read` should be defined as a static on `Delimited_Format` instead?

Even if we want to keep the above as-is, at the very least I think we should do 
```suggestion
    can_read : File -> Any
    can_read self file =
        case file.extension of
            ".csv" -> self.clone delimiter=','
            ".tsv" -> self.clone delimiter='\t'
            _ -> Nothing
```

(I know that `clone` does not currently support overridding the `delimiter` but I think we can easily adapt it).

Right now we can pass a `Delimited` instance with some more complex configuration, ask it "can you read X" and we get back a new instance - with all of our configuration wiped out. So shall we still use the original one? But the delimiter might have changed, so we probably shouldn't. But the new one does not have all of our overrides.
Of course for now we don't have such uses, because we don't use more complex configs together with `can_read` - well that just suggests me that `can_read` is sitting in wrong place and the configuration and detection are separate responsibilities that should not be coupled in this single concept.